frontend/src/TooljetDatabase/usePostgrestQueryBuilder.jsx

```diff
@@ -0,0 +1,111 @@
+import { useRef } from 'react';
+import PostgrestQueryBuilder from '@/_helpers/postgrestQueryBuilder';
+import { tooljetDatabaseService } from '@/_services';
+import { isEmpty } from 'lodash';
+import { toast } from 'react-hot-toast';
+
+export const usePostgrestQueryBuilder = ({ organizationId, selectedTable, setSelectedTableData, setTotalRecords }) => {
+  const postgrestQueryBuilder = useRef({
+    filterQuery: new PostgrestQueryBuilder(),
+    sortQuery: new PostgrestQueryBuilder(),
+    paginationQuery: new PostgrestQueryBuilder(),
+  });
+
+  const handleBuildSortQuery = (filters) => {
+    Object.keys(filters).map((key) => {
+      if (!isEmpty(filters[key])) {
+        const { column, order } = filters[key];
+        if (!isEmpty(column) && !isEmpty(order)) {
+          postgrestQueryBuilder.current.sortQuery.order(column, order);
+        }
+      }
+    });
+    updateSelectedTableData();
+  };
+
+  const updateSelectedTableData = async () => {
+    const query =
+      postgrestQueryBuilder.current.filterQuery.url.toString() +
+      '&' +
+      postgrestQueryBuilder.current.sortQuery.url.toString() +
+      '&' +
+      postgrestQueryBuilder.current.paginationQuery.url.toString();
+
+    const { headers, data, error } = await tooljetDatabaseService.findOne(organizationId, selectedTable, query);
+
+    if (error) {
+      toast.error(error?.message ?? 'Something went wrong');
+      return;
+    }
+
+    const totalRecords = headers['content-range'].split('/')[1] || 0;
+
+    if (Array.isArray(data)) {
+      setTotalRecords(totalRecords);
+      setSelectedTableData(data);
+    }
+  };
+
+  const handleBuildFilterQuery = (filters) => {
+    Object.keys(filters).map((key) => {
+      if (!isEmpty(filters[key])) {
+        const { column, operator, value } = filters[key];
+        if (!isEmpty(column) && !isEmpty(operator) && !isEmpty(value)) {
+          const currentFilterQuery = postgrestQueryBuilder.current.filterQuery.url.toString();
+
+          if (currentFilterQuery.includes(column)) {
+            const filterQuery = new PostgrestQueryBuilder();
+            filterQuery.filter(column, operator, value);
+            postgrestQueryBuilder.current.filterQuery = filterQuery;
+            return;
+          }
```

Comment:
This can be removed as there might be a case of filtering on multiple columns 
ex: id equals 9 && id not equals 10 return all rows that match